### PR TITLE
Don't decompile if a file is too large

### PR DESCRIPTION
**Pull request by riknoll**
Limit the decompilation to 1000 blocks and bail out if we hit that limit.

too_large

### CODE REVIEW

File: pxtcompiler/emitter/decompiler.ts [hunk 1]
```diff
@@ -343,6 +372,40 @@ ${output}</xml>`;
             result.success = false;
         }
 
+        function countBlock() {
+            emittedBlocks ++;
+            if (emittedBlocks > MAX_BLOCKS) {
+                let e = new Error(Util.lf("Could not decompile because the script is too large"));
```
Comment (pelikhan) [inline pxtcompiler/emitter/decompiler.ts]: Maybe just "There are too many blocks!"

Comment (riknoll) [reply pxtcompiler/emitter/decompiler.ts]: Actually, the user won't see that text. I'll remove the call to lf

File: pxtcompiler/emitter/decompiler.ts [hunk 2]
```diff
@@ -975,45 +1011,35 @@ ${output}</xml>`;
...
-                        inputs: [{
-                            kind: "value",
-                            name: "VALUE",
-                            value: negateNumericNode(n.right),
-                            shadowType: numberType
-                        }],
-                        fields: [getField("VAR", getVariableName(n.left as ts.Identifier))]
-                    };
+                    const r = mkStmt("variables_change");
+                    countBlock();
```
Comment (pelikhan) [inline pxtcompiler/emitter/decompiler.ts]: why extra countBlock?

File: webapp/src/monaco.tsx [hunk 3]
```diff
@@ -138,7 +138,9 @@ export class Editor extends srceditor.Editor {
                         .then(resp => {
                             if (!resp.success) {
                                 this.currFile.diagnostics = resp.diagnostics;
-                                return failedAsync(blockFile);
+                                let tooLarge = false;
+                                resp.diagnostics.forEach(d => tooLarge = (tooLarge || d.code === 9266 /* error code when script is too large */));
```
Comment (pelikhan) [inline webapp/src/monaco.tsx]: define constant for 9266?

**Pull request merged**